bitnami/mongodb-sharded/templates/mongos/mongos-service-per-replica.yaml

```diff
@@ -0,0 +1,49 @@
+{{- if and .Values.mongos.useStatefulSet .Values.mongos.servicePerReplica.enabled -}}
+{{- range $i := until (.Values.mongos.replicas | int) }}
+{{- $copy := deepCopy $ -}}
+{{- $context := merge $copy (dict "index" $i) -}}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "mongodb-sharded.serviceName" $ }}-{{ $i }}
+  labels: {{ include "common.labels.standard" $ | nindent 4 }}
+    app.kubernetes.io/component: mongos
+  annotations: {{- include "common.tplvalues.render" (dict "value" $.Values.mongos.servicePerReplica.annotations "context" $context) | nindent 4 }}  
+spec:
+  type: {{ $.Values.mongos.servicePerReplica.type }}
+  {{- if and $.Values.mongos.servicePerReplica.loadBalancerIP (eq $.Values.mongos.servicePerReplica.type "LoadBalancer") }}
+  loadBalancerIP: {{ $.Values.mongos.servicePerReplica.loadBalancerIP }}
+  {{- end }}
+  {{- if and (eq $.Values.mongos.servicePerReplica.type "LoadBalancer") $.Values.mongos.servicePerReplica.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges:
+  {{ with $.Values.mongos.servicePerReplica.loadBalancerSourceRanges }}
+{{ toYaml . | indent 4 }}
+  {{- end }}
+  {{- end }}
+  {{- if and (eq $.Values.mongos.servicePerReplica.type "ClusterIP") $.Values.mongos.servicePerReplica.clusterIP }}
+  clusterIP: {{ $.Values.mongos.servicePerReplica.clusterIP }}
+  {{- end }}
+  ports:
+    - name: mongodb
+      port: {{ $.Values.mongos.servicePerReplica.port }}
+      targetPort: mongodb
+      {{- if $.Values.mongos.servicePerReplica.nodePort }}
+      nodePort: {{ $.Values.mongos.servicePerReplica.nodePort }}
+      {{- else if eq $.Values.mongos.servicePerReplica.type "ClusterIP" }}
+      nodePort: null
```

Comment:
I think it is not needed to set it to `null`, would be enough without adding it for this case

Reply:
Same as above, this is the same implementation as in the service.